scripts/config.js

```diff
@@ -0,0 +1,401 @@
+        `);
+        });
+    })
+    .catch(err => {
+      console.log(chalk.red(`An error occurred due to: ${err.message}`));
+    });
+};
+
+/**
+ * Create new .env file using .env-template
+ */
+const createEnvFile = () => {
+  fs.copyFileSync('./.env-template', './.env', err => {
+    if (err) throw err;
+  });
+};
+
+/**
+ * Get the saved values from .env file so they can be used as a default values
+ *
+ * @return array containing all keys and values saved to .env file
+ */
+const findSavedValues = () => {
+  const savedEnvFile = fs.readFileSync('./.env').toString();
+
+  const settings = savedEnvFile.split('\n').reduce((obj, line) => {
+    const splits = line.split('=');
+    const key = splits[0];
+    if (splits.length > 1) {
+      obj[key] = splits.slice(1).join('=');
+    }
+    return obj;
+  }, {});
+
+  return settings;
+};
+
+/**
+ * Read all lines from existing .env file to array.
+ * If line matches one of the keys in user's answers update add value to that line.
+ * Otherwise keep the original line.
+ *
+ * @param {array} answers array containing all the inputs user has given to questions
+ *
+ * @return returns a Promise that resolves with values object containing user input and read lines
+ */
+
+const readLines = answers => {
+  return new Promise((resolve, reject) => {
+    const rl = readline.createInterface({
+      input: require('fs').createReadStream('./.env'),
+    });
+
+    const lines = [];
+    rl.on('line', function(line) {
+      lines.push(line);
+    });
+
+    rl.on('close', () => {
+      const values = { answers, lines };
+      resolve(values);
+    });
+  });
+};
+
+/**
+ *
+ * Creates array of data that is saved to .env file
+ *
+ * @param {object} values contais lines read from .env file and answers from the user
+ *
+ * @returns array containing the data that needs to be saved to .env file
+ */
+const getData = values => {
+  const { lines, answers } = values;
+
+  const data = [];
+  lines.map(line => {
+    const key = checkIfSameLine(answers, line);
+    if (key) {
+      const value = `${answers[key]}`;
+      data.push(`${key}=${value.trim()}\n`);
+    } else {
+      data.push(`${line}\n`);
+    }
+  });
+
+  return data;
+};
+
+/**
+ * Maps trough array of lines to create contet that is written to .env file. This will overwrite the previous content.
+ *
+ * @param {array} data arry of lines to be written into .env file
+ */
+const updateEnvFile = data => {
+  let content = '';
+  data.map(line => {
+    content = content + line.toString();
+  });
+
+  fs.writeFileSync('./.env', content);
+};
+
+/**
+ * Check if line contains the key from the user answers.
+ *
+ * @param {array} answers array of answers user has given to questions
+ * @param {string} line line from the existing .env file
+ *
+ * @returns key if the key matches to the one in the line
+ */
+const checkIfSameLine = (answers, line) => {
+  let foundKey;
+  if (answers) {
+    Object.keys(answers).map(key => {
```

Comment:
Instead of `map` maybe [find](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/find) could suit here better.

Another observation is that in the function docstring the type of `answers` is `array` but here `Object.keys` is invoked on it which indicates that it would be a normal `Object`. A sidenote here that in JavaScript array is a kind of object that provides some list-like functionalities and uses indices as keys. So calling `Object.keys` on an array is totally fine but it just returns the indices of the elements in the array:
```js
Object.keys(["foo", "bar", "xyz"]); // => [0, 1, 2]
```

Reply:
Indeed there was a mistake in the docstring and `answers` actually is `Object` and not `array` otherwise I think Object.keys wouldn't have worked in this case 🤔